libsplinter/src/admin/client/event/ws/actix_web_client.rs

```diff
@@ -303,6 +307,32 @@ impl AdminServiceEventClient for AwcAdminServiceEventClient {
         evt_result
             .map_err(|e| NextEventError::InternalError(InternalError::from_source(Box::new(e))))
     }
+
+    fn wait_for(
```

Comment:
This function is implementation-agnostic, so it can be pulled up to the trait as a default implementation. Any implementer that wants to override it can do so.